boards/shields/nxp_m2_wifi_bt/nxp_m2_wifi_bt.overlay

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2025 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/ {
+	chosen {
+		zephyr,bt-hci = &bt_hci_uart;
+	};
+};
+
+&m2_hci_bt_uart {
+	current-speed = <115200>;
+	status = "okay";
+
+	bt_hci_uart: bt_hci_uart {
+		compatible = "zephyr,bt-hci-uart";
+		status = "okay";
+
+		m2_bt_module {
+			compatible = "nxp,bt-hci-uart";
+			hci-operation-speed = <3000000>;
+			hw-flow-control;
+			fw-download-primary-speed = <115200>;
+			fw-download-secondary-speed = <3000000>;
+			fw-download-secondary-flowcontrol;
+		};
+	};
+};
+
+&usdhc1 {
+	/* Use dat3 detection method */
+	detect-dat3;
+	pinctrl-4 = <&pinmux_usdhc1_dat3_nopull>;
```

Comment:
Agreed like Nirav said the shield DTS should not have HW config specific to the base board.  Consider that this HW module can be plugged into other base boards instead of the RT1060 EVK.  A different board may use a different name for the SDHC interface, and different pins.

Put the base board specific details like `usdhc1` and these pins in the base board overlay for the shield.